9-regular-expressions/17-regexp-methods/article.md

```diff
@@ -254,56 +255,56 @@ let result;
 
 while (result = regexp.exec(str)) {
   alert( `Found ${result[0]} at position ${result.index}` );
-  // Found JavaScript at position 11, then
-  // Found javascript at position 33
+  // Se encontró JavaScript en la posición 11, luego
+  // Se encontró javascript en la posición 33
 }
 ```
 
-This works now as well, although for newer browsers `str.matchAll` is usually more convenient.
+Esto también, aunque para navegadores modernos `str.matchAll` es lo más conveniente.
 
-**We can use `regexp.exec` to search from a given position by manually setting `lastIndex`.**
+**Podemos usar `regexp.exec` para buscar desde una posición dada configurando manualmente el `lastIndex`.**
 
-For instance:
+Por ejemplo:
 
 ```js run
-let str = 'Hello, world!';
+let str = 'Hola, mundo!';
 
-let regexp = /\w+/g; // without flag "g", lastIndex property is ignored
-regexp.lastIndex = 5; // search from 5th position (from the comma)
+let regexp = /\w+/g; // sin la bandera "g", la propiedad `lastIndex` es ignorada
+regexp.lastIndex = 5; // buscar desde la 5ta posición (desde la coma)
 
-alert( regexp.exec(str) ); // world
+alert( regexp.exec(str) ); // mundo
 ```
 
-If the regexp has flag `pattern:y`, then the search will be performed exactly at the  position `regexp.lastIndex`, not any further.
+Si la expresión regular tiene la bandera `pattern:y`, entonces la busqueda se realizará exactamente en la posición del `regexp.lastIndex`, no más adelante.
 
-Let's replace flag `pattern:g` with `pattern:y` in the example above. There will be no matches, as there's no word at position `5`:
+Vamos a reemplazar la bandera `pattern:g` con `pattern:y` en el ejemplo anterior. No habrá coincidencias, ya que no hay palabra en la posiicón `5`:
 
 ```js run
 let str = 'Hello, world!';
 
 let regexp = /\w+/y;
-regexp.lastIndex = 5; // search exactly at position 5
+regexp.lastIndex = 5; // buscar exactamente en la posición 5
 
 alert( regexp.exec(str) ); // null
 ```
 
-That's convenient for situations when we need to "read" something from the string by a regexp at the exact position, not somewhere further.
+Es conveniente para situaciones cuando necesitamos "leer" algo de la cadena por una expresión regular en una posición exacta. That's convenient for situations when we need to "read" something from the string by a regexp at the exact position, no en otro lugar.
```

Comment:
hard to read...
ademas del mix ingles

Es conveniente para situaciones cuando, con una expresión regular, necesitamos "leer" algo de la cadena en una posición exacta, no en otro lugar.

Es conveniente para situaciones cuando necesitamos "leer" algo de la cadena en una posición exacta, no en otro lugar, , con una expresión regular.

en ingles esta feo. fuera.


Esto es conveniente cuando necesitamos "leer" algo de la cadena en una posición exacta, no en otro lugar, con una expresión regular.

```suggestion
Esto es conveniente cuando con una expresión regular necesitamos "leer" algo de la cadena en una posición exacta, no en otro lugar.
```
o con comas
Esto es conveniente cuando, con una expresión regular, necesitamos "leer" algo de la cadena en una posición exacta, no en otro lugar.

y si me apuras quito "no en otro lugar" coma incluida.  Exacta es exacta. y como cierra la oración, queda clarísima.

Esto es conveniente cuando con una expresión regular necesitamos "leer" algo de la cadena en una posición exacta.

uf 
mejor me voy a dormir